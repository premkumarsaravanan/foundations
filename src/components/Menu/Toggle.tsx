import * as React from 'react'

export interface ToggleProps {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  isChecked: boolean
  isResponsive: boolean
}

export const Toggle: React.SFC<ToggleProps> = ({ onChange, isChecked, isResponsive = true }) => (
  <>
    <label className={`toggle ${isResponsive ? 'is-responsive' : ''}`}>
      <input type="checkbox" onChange={onChange} checked={isChecked} />
      <div>
        <div>
          <span />
          <span />
        </div>
        <svg>
          <use xlinkHref="#path" />
        </svg>
        <svg>
          <use xlinkHref="#path" />
        </svg>
      </div>
    </label>
    <svg xmlns="http://www.w3.org/2000/svg" style={{ display: 'none' }}>
      <symbol xmlns="http://www.w3.org/2000/svg" viewBox="0 0 44 44" id="path">
        <path d="M22,22 L2,22 C2,11 11,2 22,2 C33,2 42,11 42,22" />
      </symbol>
    </svg>
  </>
)
