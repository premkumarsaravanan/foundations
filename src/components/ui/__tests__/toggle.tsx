import * as React from 'react'
import { shallow } from 'enzyme'
import Toggle from '../toggle'
import toJson from 'enzyme-to-json'

describe('Toggle', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<Toggle isChecked onChange={jest.fn()} />))).toMatchSnapshot()
  })

  it('responds to an onChange event', () => {
    const onChange = jest.fn()
    const wrapper = shallow(<Toggle isChecked onChange={onChange} />)
    wrapper
      .find('input')
      .first()
      .simulate('change')
    expect(onChange).toBeCalledTimes(1)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
