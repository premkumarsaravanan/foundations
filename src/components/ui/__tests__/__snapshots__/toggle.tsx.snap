// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should match a snapshot 1`] = `
<Fragment>
  <label>
    <input
      checked={true}
      onChange={[MockFunction]}
      type="checkbox"
    />
    <div>
      <div>
        <span />
        <span />
      </div>
      <svg>
        <use
          xlinkHref="#path"
        />
      </svg>
      <svg>
        <use
          xlinkHref="#path"
        />
      </svg>
    </div>
  </label>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="path"
      viewBox="0 0 44 44"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M22,22 L2,22 C2,11 11,2 22,2 C33,2 42,11 42,22"
      />
    </symbol>
  </svg>
</Fragment>
`;
